src/makeCalculator.js

```diff
@@ -4,7 +4,45 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      if (number === 0) {
+        throw new Error('Cannot divide by zero');
+      }
+
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
```

Comment:
The `reset` method correctly resets the `result` to 0 and returns `this` for method chaining, which is a requirement for the task.